src/Microsoft.TestPlatform.Extensions.BlameDataCollector/MiniDumpWriteDump.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.TestPlatform.Extensions.BlameDataCollector;
+
+internal enum MiniDumpTypeOption
+{
+    // This is a copy of DumpTypeOption, we need both of those types, because this
+    // dumper is included as file in both MiniDumpTool, and BlameDataCollector, and
+    // blame references MiniDumpTool, so we get the enum defined twice otherwise.
+    Full,
+    WithHeap,
+    Mini,
+}
+
+internal class MiniDumpWriteDump
+{
+    public static void CollectDumpUsingMiniDumpWriteDump(Process process, string outputFile, MiniDumpTypeOption type)
+    {
+        // Open the file for writing
+        using var stream = new FileStream(outputFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None);
+        NativeMethods.MinidumpExceptionInformation exceptionInfo = default;
+
+        NativeMethods.MinidumpType dumpType = NativeMethods.MinidumpType.MiniDumpNormal;
+        switch (type)
```

Comment:
[minor] Could you refactor (VS refactoring) this switch to be a switch expression? It's easier to spot that it initializes/touches only 1 thing.